pulp_rpm/app/management/commands/rpm-datarepair.py

```diff
@@ -22,9 +23,21 @@ def handle(self, *args, **options):
 
         if issue == "2460":
             self.repair_2460()
+        elif issue == "2258":
+            self.repair_2258()
         else:
             raise CommandError(_("Unknown issue: '{}'").format(issue))
 
+    def repair_2258(self):
+        """Re-examine package header and store signing key for issue #2258."""
+        packages = Package.objects.filter(signer_key_id__isnull=True)
+        for package in packages:
```

Comment:
Use `.iterator()` here otherwise the query will use a ton of memory

Reply:
But also - consider rewriting this query in a different way to avoid as many lookups.  As RPM packages have only one artifact, it is possible to use `ContentArtifact` as a middleman that allows you to access fields of both the `Package` and `Artifact` tables simultaneously via JOIN.

Example here: https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/tasks/publishing.py#L474-L495

Of course since this is just a data repair, maybe simple is better.  @ggainey do you have an opinion?

Reply:
I've added `iterator()` and the warning. My preference here in situations like this is always simplicity / clarity.

The slow part is going to be downloading the RPM anyway.

Reply:
Well `.iterator()` is pretty simple, it just skips putting the data in Django's cache if you are only using it once.  It's not a speed concern but a memory one since for large querysets this really adds up.  `Package` objects are pretty large relative to the normal stuff you might use Django for.

Reply:
Yes, I wasn't complaining about `iterator()`, I was responding to the other suggesting of doing a custom join.